astropy/io/fits/tests/test_connect.py

```diff
@@ -68,12 +67,19 @@ def setup_class(self):
         )
 
     def test_overwrite_with_path(self, tmp_path):
-        filename = "temp.fits"
+        filename1 = tmp_path / "temp1.fits"
         t1 = Table(self.data)
-        with contextlib.chdir(tmp_path):
-            t1.write(filename, format="fits")
-            t1.write(Path(filename), format="fits", overwrite=True)
-        t1.write(Path(tmp_path / filename), format="fits", overwrite=True)
+        t1.write(filename1, format="fits")
+        ref_mod_time1 = filename1.stat().st_mtime
+        t1.write(filename1, format="fits", overwrite=True)
+        assert filename1.stat().st_mtime > ref_mod_time1
+
+        with chdir(tmp_path):
+            filename2 = filename1.with_stem("temp2").relative_to(tmp_path)
```

Comment:
Why make it so complicated ? The test was fine (maybe just missing a comment to explain why using chdir) and other cases must already be tested elsewhere. Can you revert to the original version and add the timestamp check ?

Reply:
I'm actually more worried that the test is failing on Windows now (possibly not in a reproducible fashion). I'll avoid modifying the test any further until I have an explanation for this failure, so we don't accidently hide a bug under the carpet.